src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpImportServiceDAO.scala

```diff
@@ -45,12 +53,20 @@ class HttpImportServiceDAO(implicit val system: ActorSystem, implicit val materi
         }
       case otherResp =>
         // see if we can extract errors
-        val responseString = otherResp.entity match {
-          case HttpEntity.Strict(_, data) => data.utf8String
-          case _ => otherResp.toString()
+        val responseStringFuture = otherResp.entity match {
+          case HttpEntity.Strict(_, data) =>
+            Future.successful(data.utf8String)
+          case nonStrictEntity =>
+            nonStrictEntity.toStrict(10.seconds).map(_.data.utf8String)
```

Comment:
calling `toStrict` can be a memory hog for very large responses. In this case, we know that the responses from import service are sane and small, so this feels ok.